skipruntime-ts/core/src/internals/skipruntime_module.ts

```diff
@@ -1386,111 +1386,128 @@ class Manager implements ToWasmManager {
   };
 }
 
-interface Timeout {
-  unref: () => void;
+/** Min heap-based priority queue whose keys are numbers and values are arrays. */
+class MinArrayPriorityQueue<T> {
+  private heap: [priority: number, values: T[]][] = [];
+  private priorityIndex = new Map<number, number>();
+
+  insert(priority: number, value: T): void {
+    const currentIndex = this.priorityIndex.get(priority);
+    if (currentIndex !== undefined) {
+      this.heap[currentIndex][1].push(value);
+    } else {
+      let i = this.heap.length;
+      this.heap.push([priority, [value]]);
+      this.priorityIndex.set(priority, i);
+      while (i > 0) {
+        const parent = (i - 1) >> 1;
+        if (this.heap[i][0] >= this.heap[parent][0]) break;
+        this.swap(i, parent);
+        i = parent;
+      }
+    }
+  }
+
+  peekPriority(): number | undefined {
+    return this.heap[0]?.[0];
+  }
+
+  popValues(): T[] | undefined {
+    const top = this.heap[0];
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```

Comment:
Is this really needed?

Reply:
We have enabled `reportUnusedDisableDirectives: "error",` so we would get an error if it wasn't needed

Reply:
It comes from `"noUncheckedIndexedAccess": false,` resulting in `top` not being `undefined`

Reply:
I thought we had enabled `noUncheckedIndexedAccess`, I'll look into it.